Tests/tests/test_SystemAudioPlayer.cpp

```diff
@@ -0,0 +1,1356 @@
+            _T("\"layout\": \"interleaved\"},")
+            _T("\"websocketsecparam\": {")
+                _T("\"cafilenames\": [")
+                    _T("{\"cafilename\": \"/etc/ssl/certs/Xfinity_Subscriber_ECC_Root.pem\"}")
+                _T("],")
+                _T("\"certfilename\": \"...\",")
+                _T("\"keyfilename\": \"...\"}")
+            _T("}"),
+        response
+        ));
+
+        EXPECT_EQ(response, _T("{\"success\":false}"));
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+
+/**
+ * @name  : SAPConfigWithoutOpen
+ * @brief : Sets a config without player instance
+ * 
+ * @param[in]   :  audiotype ,sourcetype, playmode
+ * @return      :  {success: false}
+ */
+
+TEST_F(SAPInitializedTest,SAPConfigWithoutOpen) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+    _T("config"),
+    _T("{\"id\": 1, "
+       "\"pcmconfig\": {\"format\": \"S16LE\", \"rate\": \"22050\", \"channels\": \"1\", \"layout\": \"interleaved\"}, "
+       "\"websocketsecparam\": {\"cafilenames\": [{\"cafilename\": \"/etc/ssl/certs/Xfinity_Subscriber_ECC_Root.pem\"}], "
+       "\"certfilename\": \"...\", \"keyfilename\": \"...\"}}"),
+    response
+    ));
+
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+}
+
+/*******************************************************************************************************************
+ * Test function for play
+ * Open                    :
+ *                Plays audio on the specified player.
+ *
+ *                @return Response object success status
+ * Use case coverage:
+ *                @Success : 1
+ *                @Failure : 3
+ ********************************************************************************************************************/
+
+/**
+ * @name  : SAPPlay
+ * @brief : plays a player instance
+ * 
+ * @param[in]   :  id ,url 
+ * @return      :  {success: true}
+ */
+
+TEST_F(SAPInitializedTest, SAPPlay) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"websocket\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    // Find the position of "id" in the response
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+        // Find the start and end positions of the id value
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("play"), _T("{\"id\": ") + std::to_string(speechId) + _T(", \"url\": \"https://www.google.com\" }"), response));
```

Comment:
how would google page provide data for the play (that too when websocket is configured as source type)? Shouldn't the test return failure?